### ISSUE TITLE
[Safari webkit] "Maximum call stack exceeded" - When importing CSV

### ISSUE THREAD

**Issue by johan-lejdung** [actualbudget/actual#21]
This line seems to be the issue. https://github.com/actualbudget/actual/blob/83cab80633b5dbcc3566e6292528dbe5f462a80a/packages/loot-core/src/server/sync/index.js#L131

When I upload the following file and try to import you get the error
[csvimport2.csv](https://github.com/actualbudget/actual/files/8599446/csvimport2.csv)

<img width="804" alt="Screenshot 2022-05-01 at 19 57 44" src="https://user-images.githubusercontent.com/9266884/166158567-a02f8208-9db9-497e-8b13-91b72f6194a9.png">
<img width="564" alt="Screenshot 2022-05-01 at 19 57 48" src="https://user-images.githubusercontent.com/9266884/166158571-c0ccbac8-ffc1-4555-a1e1-ed3710de5835.png">
<img width="1727" alt="Screenshot 2022-05-01 at 19 58 15" src="https://user-images.githubusercontent.com/9266884/166158572-7066af05-80c5-4b88-ab1e-5142d26363fe.png">

I might try to fix this at some point, but thought it would be good to post here for documentation sake.

**Comment by johan-lejdung** [comment]
Interesting this seems to be a Safari issue.. as suggested over here: https://github.com/mono/mono/issues/15981

But the code should probably be adjusted to not grow too large in query still.

I can confirm this is not an issue on Chrome, reported on Safari Version 15.4 (17613.1.17.1.13)

**Comment by jtbayly** [comment]
Hmm... this is what I ran into that stopped me from actually trying out Actual. I'd love to see it fixed. I had no idea I'd be able to work around the problem with another browser.

**Comment by j-f1** [comment]
I believe this was fixed by my commit https://github.com/actualbudget/actual/pull/457/commits/d868645d40fbc6105fe8b1d1a48e93b03c7a4c27 which is included in the latest release. Can you try it out and let me know if it works better?

**Comment by johan-lejdung** [comment]
@j-f1 how do I make sure I have that commit? 

I'm on the master branch and have done `git pull` but I cannot see that commit hash anywhere:

```
56e5c33c3eac7f645c1c58d6a72df2975fa08760 (HEAD -> master, origin/master, origin/HEAD) Re-generate nordigen token (#156)
37c5278abfb1e239fe301e21d22129d015f49c16 (tag: v23.3.1) <F0><9F><94><96> (23.3.1) fix nordigen usage in fly.io (#155)
d1482e38fe3f24f136174dd57f008fb0993145c7 (tag: v23.3.0) Remove extraneous $
1cce6137a018a2de808b6513755910e4872244b1 :bookmark: (23.3.0) experimental bank-sync, maintenance, fixes (#154)
27ea0ddd98eb4606d208e84e1d786db03cdda5fa Fix Docker actions failing on PRs from forks (#153)
bd7992a45ac7a806431ec94fa7021fefffabe6e5 :rotating_light:  adding trailing commas everywhere (#148)
95a0769eaa486dc74bb9b923f6e7898fa500b7fc Fix edge image tagging (#149)
af9713d5b165e270fcdd6a0c2022b13474ee4e60 :sparkles: (bank-sync) meaningful default balance calculation (#144)
ab9f77182989099b15678babd37d788bf2d04f64 Improve edge image build times (#147)
fa0c91b22c7c3e66ec2f5cc0461520ed6b3ceba8 :arrow_up:  upgrade yarn to v3.4.1 and add interactive-tools plugin (#146)
2e5f5b48fae7291340974a13ff2ad4984dbcb484 :sparkles: (bank-sync) expose available banks via an endpoint (#145)
a52ae2dba106a160e392c3dfe8bee7b84d5f829b Update the :edge images to use the very latest web UI version (#121)
19cd163b30f2ae021e743b3f5d1dd95bc939c290 Backend integration with Nordigen - account sync (#74)
06b687e8997472b7b5bdc150b5b60bb6a11e8bfe Auto-configure the client<E2><80><99>s server URL (#135)
fa42c2202ef936b6a63df0c4727b2bfa1b0008ec Replace require with import (#133)
0537544de27af650fcb25b995315f50a24fba6d1 (tag: v23.2.9) :bookmark: (23.2.9) ability to ignore SharedArrayBuffer error (#132)
be1c1197998b87ed7b48579b0cb2286040e2b5f5 Upgrade to ESM, update to latest dependencies (#128)
76f398dd16eaafc348fbc132f796da2513f86059 Move source code to an src/ directory (#131)
064d3e5dbb2e99d0c197df86478603c8b0da51cc (tag: v23.2.6) :bookmark: (23.2.6) improved HTTPs configurability (#129)
c6c324323476d8e93b3fe9130d1227efb402d554 Replace '\n' in key/cert text with literal newlines (#127)
60b183b0946a631a3e16d1deff704747f945e5e0 Fix / route returning the wrong MIME type (#125)
9b5fb43ad376c179c111bca996979302a92ada49 Improve the server config (#120)
7c58261b7296408cd79ff96d010a62d6a1f5d652 (tag: v23.2.5) :bookmark: (23.2.5) security focused release (#122)
712ca71656b4f47b507b228e1bde81e2c49eb975 :arrow_up:  upgrade express*, bcrypt and body-parser (#119)
```

**Comment by j-f1** [comment]
It’s part of https://github.com/actualbudget/actual/pull/457 — you may have to run `yarn install` to get the latest version of the frontend if you’re running locally using `yarn`.

**Comment by johan-lejdung** [comment]
You are correct! I'll close this :) Thank you. 

Another interesting thing happened in the update though, I were no longer able to open my old finances (fine by me since I didn't have too many things in there and can just re-import) with the error `Error: file does not exist: /app/user-files/file-a87d41c5-6360-41f6-a690-ac6eaa4ceb92.blob`. 

I looked at the folder structure and I saw this:
```
> ls user-files/
a87d41c5-6360-41f6-a690-ac6eaa4ceb92  file-a12b084d-5c74-4342-abce-ed8e3e29294b.blob  group-bf81abe8-927e-4bf0-9a8c-2fa9ae1a43e8.sqlite
ls user-files/a87d41c5-6360-41f6-a690-ac6eaa4ceb92
backups  cache.sqlite  db.sqlite  metadata.json
```

It seems the application has changed storage structure from folder to `.blob` files. Is there a way to migrate?
